mods/d2k/rules/infantry.yaml

```diff
@@ -267,13 +267,26 @@ saboteur:
 		Flashes: 0
 		EnterBehaviour: Suicide
 	-RevealOnFire:
+	GrantConditionOnDeploy:
+		DeployedCondition: deployed
+		DeploySound: STEALTH1.WAV
+		UndeployOnCondition: decharged
+		PauseOnCondition: !charged
+	GrantConditionAfterDelay@CHARGE:
+		Delay: 1250
+		Condition: charged
+		RequiresCondition: !deployed
+	GrantConditionAfterDelay@DECHARGE:
+		Delay: 375
+		Condition: decharged
+		RequiresCondition: deployed
 	Cloak:
-		InitialDelay: 85
+		InitialDelay: 0
 		CloakDelay: 85
-		CloakSound: STEALTH1.WAV
-		UncloakOn: Attack, Unload, Infiltrate, Demolish, Move, Damage, Heal
+		UncloakOn: Attack, Unload, Infiltrate, Demolish, Damage, Heal
 		IsPlayerPalette: true
 		PauseOnCondition: cloak-force-disabled
+		RequiresCondition: deployed
```

Comment:
I'm not sure if this is still a controversial request, but IMO prefer `RequiresCondition` / `PauseOnCondition` at the top of trait definitions.